extensions/transfer-functions/transfer-functions-core/src/main/java/org/eclipse/dataspaceconnector/transfer/functions/core/flow/http/HttpFunctionConfiguration.java

```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright (c) 2021 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.transfer.functions.core.flow.http;
+
+import okhttp3.OkHttpClient;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * Configuration for the HTTP data flow controller and status checker.
+ */
+public class HttpFunctionConfiguration {
+    private String transferEndpoint;
+    private String checkEndpoint;
+    private Set<String> protocols;
+    private Supplier<OkHttpClient> clientSupplier;
```

Comment:
Why not directly use the `OkHttpClient`? It is resolvable via the `ServiceExtensionContext`

Reply:
I need to document this...The extension will allow the use of custom interceptors to set headers for things like authorization.